src/components/Articles.jsx

```diff
@@ -1,10 +1,39 @@
 import { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { getArticles } from "../utils/api";
 
 export const Articles = () => {
-    const [showArticles, setShowArticles] = useState([]);
+  const [articles, setArticles] = useState([]);
+  const [err, setErr] = useState(null);
+  const { topic } = useParams();
 
-    useEffect(() => {
-        
-    })
+  useEffect(() => {
+    getArticles(topic)
+      .then((articlesFromApi) => {
+        setArticles(articlesFromApi);
+        setErr(null);
+      })
+      .catch((err) => {
```

Comment:
Good to see error handling however you need to think about the behaviour of the BE. In the case of a topic with no articles, we don't get an error, simply an empty array. You will need to handle this case separately but still retain the catch block to handle other errors - a more appropriate error message will be needed here.